pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -210,11 +210,19 @@ func (c *NPLController) enqueueSvcUpdate(oldObj, newObj interface{}) {
 		} else if newSvcAnnotation == "true" {
 			podKeys = sets.NewString(c.getPodsFromService(newSvc)...)
 		}
-	} else if !reflect.DeepEqual(oldSvc.Spec.Selector, newSvc.Spec.Selector) {
+	}
+
+	var oldPodSet, newPodSet sets.String
+	if oldSvc.Spec.Type != newSvc.Spec.Type {
+		newPodSet = sets.NewString(c.getPodsFromService(newSvc)...)
```

Comment:
What Service types we care about? In all the Service type change case, we always use the newPodSet?

Reply:
If nplAnnotation changes from false to true, the newPortSet is already counted in, and we can avoid getting the set twice?